### PR TITLE
[Bug]: fix type issues in projects

### LINKED ISSUE #976: [Type Error]: `node_modules/solid-start/data/createRouteAction.tsx:6:32 - error TS2307: Cannot find module 'server/server-functions/types' or its corresponding type declarations.`
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Noticed this upon upgrading to all latest solid deps

> node_modules/solid-start/data/createRouteAction.tsx:6:32 - error TS2307: Cannot find module 'server/server-functions/types' or its corresponding type declarations.

I can hide this with `skipLibCheck: true` in tsconfig.json

ScreenShot 2023-08-09 at 02 01 57@2x

### Expected behavior 🤔

nice to have: no type errors in installed packages

### Steps to reproduce 🕹

```
 npm init solid@latest
Need to install the following packages:
  create-solid@0.3.1
Ok to proceed? (y) y

create-solid version 0.2.30

Welcome to the SolidStart setup wizard!

There are definitely bugs and some feature might not work yet.
If you encounter an issue, have a look at https://github.com/solidjs/solid-start/issues and open a new one, if it is not already tracked.

✔ Where do you want to create the app? … .
✔ Which template do you want to use? › bare
✔ Server Side Rendering? … yes
✔ Use TypeScript? … yes
found matching commit hash: 2f44d456e4814b997058fb366d7361f0d6b1e4e8
/Users/drhode23/.degit/github/solidjs/solid-start/2f44d456e4814b997058fb366d7361f0d6b1e4e8.tar.gz already exists locally
extracting /examples/bare from /Users/drhode/.degit/github/solidjs/solid-start/2f44d456e4814b997058fb366d7361f0d6b1e4e8.tar.gz to /Users/drhode/project/packages/mf-core/mfcore2/.solid-start
cloned solidjs/solid-start#main to /Users/drhode/project/packages/mf-core/mfcore2/.solid-start
✔ Copied project files

Next steps:
  1: npm install
  2: npm run dev -- --open

To close the dev server, hit Ctrl-C
mfcore2 ➤ npm i

added 380 packages, and audited 381 packages in 15s

50 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities


# Add a script to package.json `"type-check": "tsc --noEmit"`


mfcore2 ➤ npm run type-check

> type-check
> tsc --noEmit

node_modules/solid-start/data/createRouteAction.tsx:6:32 - error TS2307: Cannot find module 'server/server-functions/types' or its corresponding type declarations.

6 import { ServerFunction } from "server/server-functions/types";
                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/solid-start/vite/plugin.d.ts:23:18 - error TS7016: Could not find a declaration file for module '../fs-router/router'. '/Users/drhode23/packages/mf-core/mfcore2/node_modules/solid-start/fs-router/router.js' implicitly has an 'any' type.

23   router: import("../fs-router/router").Router;
                    ~~~~~~~~~~~~~~~~~~~~~


Found 2 errors in 2 files.

Errors  Files
     1  node_modules/solid-start/data/createRouteAction.tsx:6
     1  node_modules/solid-start/vite/plugin.d.ts:23
mfcore2 ➤    
```

### Context 🔦

new projects should have no type errors.

we could probably set `skipLibCheck: true` in tsconfig if these type errors aren't meaningful.

### Your environment 🌎

> "Please paste the output of running whatever the command is for your environment."

I'm not sure what command that would be.

### PR DESCRIPTION
**Pull request by caseybaggz**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

closes #976 

Currently, the latest version of SS throws type errors due to a bad path in createRouteAction.tsx. Additionally, due to the inclusion of `solid-start/env` in types, forces users to list _all_ types they would use in a project due to how the field works.

## What is the new behavior?
- Fixes the bad path and adds `type` import alias to ensure TS removes them and also make code imports easier to read
- Updates all examples to use `global.d.ts` which references the `solid-start/env` file so the templates no longer require users to list every dep type their project adds throughout time.

## Other information

I haven't checked out the Astro branch, but we should make sure this change also makes it's way into there.

### CODE REVIEW

File: examples/hackernews/package.json [hunk 1]
```diff
@@ -9,17 +9,17 @@
   "main": "./dist/index.js",
   "devDependencies": {
     "@types/babel__core": "^7.20.1",
-    "@types/node": "^18.16.19",
+    "@types/node": "^18.17.5",
     "esbuild": "^0.14.54",
-    "solid-start-node": "^0.3.0",
+    "solid-start-node": "workspace:^",
```
Comment (edivados) [inline examples/hackernews/package.json:14]: Doesn't this lead to create-solid cli initalizing broken projects? As far as I know it pulls from the repo. Worth checking.

Comment (caseybaggz) [reply examples/hackernews/package.json:14]: @edivados good catch! This was a pnpm takeover and my mind totally defaulted to the examples shipping with the CLI. 🙈

Comment (caseybaggz) [reply examples/hackernews/package.json:14]: @edivados ready! Thanks for the patience and catch!

Comment (edivados) [reply examples/hackernews/package.json:14]: Cool. Looks like two spots are left.
- node adapter on this file
- solid-start and solid-start-node on with-vitest example

Comment (caseybaggz) [reply examples/hackernews/package.json]: @edivados donezies.

**Pull request merged**